pkg/cluster/node.go

```diff
@@ -0,0 +1,199 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/ministryofjustice/cloud-platform-cli/pkg/client"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FindNode takes a node name and returns the node object
+func (cluster *Cluster) FindNode(name string) (*v1.Node, error) {
+	var n v1.Node
+	for _, node := range cluster.Nodes {
+		if node.Name == name {
+			return &node, nil
+		}
+	}
+
+	return &n, errors.New("node not found")
+}
+
+// getAllNodes returns a slice of all nodes in a cluster
+func getAllNodes(c *client.Client) ([]v1.Node, error) {
+	n := make([]v1.Node, 0)
+	nodes, err := c.Clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	n = append(n, nodes.Items...)
+	return n, nil
+}
+
+// getOldestNode returns the oldest node in a cluster
+func getOldestNode(c *client.Client) (v1.Node, error) {
+	nodes, err := getAllNodes(c)
+	if err != nil {
+		return v1.Node{}, err
+	}
+
+	return oldestNode(nodes)
+}
+
+// oldestNode takes a slice of nodes and returns the oldest node
+func oldestNode(nodes []v1.Node) (v1.Node, error) {
+	oldestNode := nodes[0]
+	for _, node := range nodes {
+		if node.CreationTimestamp.Before(&oldestNode.CreationTimestamp) {
+			oldestNode = node
+		}
+	}
+
+	return oldestNode, nil
+}
+
+// HealthCheck ensures the cluster is in a healthy state
+// i.e. all nodes are running and ready
+func (c *Cluster) HealthCheck() error {
+	err := c.areNodesReady()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// areNodesReady checks if all nodes are in a ready state
+func (c *Cluster) areNodesReady() error {
+	for _, node := range c.Nodes {
+		if node.Status.Conditions[0].Type != "Ready" && node.Status.Conditions[0].Status == "True" {
+			return fmt.Errorf("node %s is not ready", node.Name)
+		}
+	}
+
+	return nil
+}
+
+// CompareNodes confirms if the number of nodes in a snapshot
+// is the same as the number of nodes in the cluster.
+func (c *Cluster) CompareNodes(snap *Snapshot) (err error) {
+	if len(c.Nodes) != len(snap.Cluster.Nodes) {
+		return fmt.Errorf("number of nodes are different")
+	}
+
+	return nil
+}
+
+func (c *Cluster) DeleteNode(client *client.Client, awsProfile, awsRegion string, node *v1.Node) error {
```

Comment:
'c' is not used inside the function so can it be removed?

Reply:
Agreed, will remove it.

Reply:
Removed receiver.